src/data/pnat/basic.lean

```diff
@@ -112,6 +112,13 @@ coe_injective.add_left_cancel_semigroup coe (λ _ _, rfl)
 instance : add_right_cancel_semigroup ℕ+ :=
 coe_injective.add_right_cancel_semigroup coe (λ _ _, rfl)
 
+@[priority 10]
+instance : covariant_class ℕ+ ℕ+ ((+)) (≤) :=
+⟨begin
+  rintro ⟨a, ha⟩ ⟨b, hb⟩ ⟨c, hc⟩,
+  simp [←pnat.coe_le_coe]
+end⟩
```

Comment:
I would guess
```suggestion
⟨λ ⟨a, ha⟩ ⟨b, hb⟩ ⟨c, hc⟩, by simp [←pnat.coe_le_coe]⟩
```

Reply:
```
@[priority 10]
instance : covariant_class ℕ+ ℕ+ ((+)) (≤) :=
⟨λ ⟨a, ha⟩, by { extract_goal, }⟩
```

```
example (_x : ℕ+)
  (_fun_match : ∀ (_a : ℕ+) {n₁ n₂ : ℕ+},
                  n₁ ≤ n₂ → _a + n₁ ≤ _a + n₂) (a : ℕ)
  (_mlocal._fresh.304.3605 _mlocal._fresh.304.3606 : ℕ+)
  (ha : 0 < a) :
  __mlocal__fresh_304_3605 ≤ __mlocal__fresh_304_3606 →
  ⟨a, ha⟩ + __mlocal__fresh_304_3605 ≤
    ⟨a, ha⟩ + __mlocal__fresh_304_3606 :=
begin
  admit,
end
````

I think the destruction notation doesn't work properly with implicit arguments.